monkey/monkey_island/cc/resources/configuration_import.py

```diff
@@ -0,0 +1,84 @@
+import json
+import logging
+from dataclasses import dataclass
+from json.decoder import JSONDecodeError
+
+import flask_restful
+from flask import request
+
+from common.utils.exceptions import (
+    InvalidConfigurationError,
+    InvalidCredentialsError,
+    NoCredentialsError,
+)
+from monkey_island.cc.resources.auth.auth import jwt_required
+from monkey_island.cc.services.config import ConfigService
+from monkey_island.cc.services.utils.config_encryption import decrypt_config
+
+logger = logging.getLogger(__name__)
+
+
+class ImportStatuses:
+    UNSAFE_OPTION_VERIFICATION_REQUIRED = "unsafe_options_verification_required"
+    INVALID_CONFIGURATION = "invalid_configuration"
+    PASSWORD_REQUIRED = "password_required"
+    WRONG_PASSWORD = "wrong_password"
+    IMPORTED = "imported"
+
+
+@dataclass
+class ResponseContents:
+    import_status: str = ImportStatuses.IMPORTED
+    message: str = ""
+    status_code: int = 200
+    config: str = ""
+    config_schema: str = ""
+
+    def form_response(self):
+        return self.__dict__
+
+
+class ConfigurationImport(flask_restful.Resource):
+    SUCCESS = False
+
+    @jwt_required
+    def post(self):
+        request_contents = json.loads(request.data)
+        try:
+            config = ConfigurationImport._get_plaintext_config_from_request(request_contents)
+            if request_contents["unsafeOptionsVerified"]:
```

Comment:
At this point, the configuration is sent to the front end just to verify if it contains unsafe options. 
Other options:
1. do the whole encryption on the front end (less extendable if we want to save a list of configurations on the back end in the future?)
2. implement the safety check on the back end.

Reply:
I'm still not convinced that encrypting this for the user provides the best user experience. I think just giving a checkbox that allows user to omit the sensitive information from the exported config is less complex, less maintenance, and we don't have to worry about whether or not the encryption scheme we chose is compliant with the user's security policy.

Reply:
We don't have to worry about encryption, if they want they can encrypt themselves. How will we know which information is sensitive? Did user input a real-world password or just a common password for his locale? Tracking sensitive fields is more maintenance than just encrypting the export.

Reply:
I talked to Barak. We'll proceed with the encryption. 

> At this point, the configuration is sent to the front end just to verify if it contains unsafe options.
Other options:

Can we create a #TODO and/or a github issue to address this ^